databricks/koalas/base.py

```diff
@@ -945,11 +945,12 @@ def value_counts(self, normalize=False, sort=True, ascending=False, bins=None, d
             raise NotImplementedError("value_counts currently does not support bins")
 
         if dropna:
-            sdf_dropna = self._internal._sdf.dropna()
+            sdf_dropna = self._internal._sdf.select(self._scol).dropna()
         else:
             sdf_dropna = self._internal._sdf
         index_name = SPARK_INDEX_NAME_FORMAT(0)
-        sdf = sdf_dropna.groupby(self._scol.alias(index_name)).count()
+        column_name = self._internal.data_columns[0]
+        sdf = sdf_dropna.groupby(sdf_dropna[column_name].alias(index_name)).count()
```

Comment:
Although I don't remember why we changed it, maybe we can go back to the previous implementation?
https://github.com/databricks/koalas/pull/949/files#diff-f38e425003958ac0d1bd6ffa418f2613L1892-L1897
Do you remember why we changed to use `dropna`?

Reply:
Maybe this is the reason. https://github.com/databricks/koalas/pull/949#issuecomment-548185858

Reply:
yes you're right. thanks for reminding me either :)